tests/unit/local/apigw/test_local_apigw_service.py

```diff
@@ -506,6 +512,57 @@ def test_get_current_route_keyerror(self):
         with self.assertRaises(KeyError):
             self.api_service._get_current_route(request_mock)
 
+    @patch.object(LocalApigwService, "get_request_methods_endpoints")
+    @patch("samcli.local.apigw.local_apigw_service.ServiceErrorResponses")
+    @patch("samcli.local.apigw.local_apigw_service.LocalApigwService._valid_identity_sources")
+    def test_request_contains_lambda_auth_missing_identity_sources(
+        self, validate_id_mock, service_error_mock, request_mock
+    ):
+        route = self.api_gateway_route
+        route.authorizer_object = LambdaAuthorizer("", "", "", [], "")
+
+        self.api_service._get_current_route = MagicMock()
+        self.api_service._get_current_route.return_value = route
+        self.api_service._get_current_route.methods = []
+        self.api_service._get_current_route.return_value.payload_format_version = "2.0"
+
+        mocked_missing_lambda_auth_id = Mock()
+        service_error_mock.missing_lambda_auth_identity_sources.return_value = mocked_missing_lambda_auth_id
+
+        request_mock.return_value = ("test", "test")
+
+        validate_id_mock.return_value = False
+
+        result = self.api_service._request_handler()
+
+        self.assertEqual(result, mocked_missing_lambda_auth_id)
+
+    def test_valid_identity_sources_not_lambda_auth(self):
+        route = self.api_gateway_route
+        route.authorizer_object = None
+
+        self.assertFalse(self.api_service._valid_identity_sources(route))
+
+    @parameterized.expand(
+        [
+            (True,),
+            (False,),
+        ]
+    )
+    @patch("samcli.local.apigw.authorizers.lambda_authorizer.LambdaAuthorizer._parse_identity_sources")
+    @patch("samcli.local.apigw.authorizers.lambda_authorizer.LambdaAuthorizer.identity_sources")
+    def test_valid_identity_sources_id_source(self, is_valid, id_source_prop_mock, lambda_auth_parse_mock):
+        route = self.api_gateway_route
+        route.authorizer_object = LambdaAuthorizer("", "", "", [], "")
+
+        mocked_id_source_obj = Mock()
+        mocked_id_source_obj.is_valid = Mock(return_value=is_valid)
+        route.authorizer_object.identity_sources = [mocked_id_source_obj]
+
+        # create a dummy Flask app to populate the request object with testing data
+        with flask.Flask(__name__).test_request_context():
+            self.assertEqual(self.api_service._valid_identity_sources(route), is_valid)
```

Comment:
This populates the `request` global object with testing/dummy data that Flask sets. This shouldn't change any behaviour since the `_valid_identity_sources()` does not call any methods that use the `request` global, nor does it actually use any variables in the request (the calls are mocked out).